actionview/lib/action_view/base.rb

```diff
@@ -237,11 +237,28 @@ def initialize(lookup_context, assigns, controller) # :nodoc:
       _prepare_context
     end
 
-    def _run(method, template, locals, buffer, add_to_stack: true, &block)
+    def _run(method, template, locals, buffer, add_to_stack: true, locals_disabled: false, explicit_locals: false, &block)
       _old_output_buffer, _old_virtual_path, _old_template = @output_buffer, @virtual_path, @current_template
       @current_template = template if add_to_stack
       @output_buffer = buffer
-      public_send(method, locals, buffer, &block)
+
+      if explicit_locals
+        if locals_disabled && locals.present?
+          raise ArgumentError, "#{template.short_identifier} does not accept locals."
```

Comment:
I'm not sure this branch is worth it. If we disabled locals, trying to pass them as keyword args will raise an ArgumentError right?

So we could just do that. (will just need to update the `.sub` again to handle both message `missing keyword: :bar` and `unknown keyword: :bar`)

Reply:
Without this carve-out, the error is pretty cryptic: `wrong number of arguments (given 2, expected 1)`. We could of course look for that error and swap it out, but I think this solution is more readable and less fragile.

Reply:
Really?

```
>> def foo(buf); end
=> :foo
>> foo(nil, **{bar: 1})
(irb):11:in `foo': unknown keyword: :bar (ArgumentError)
```

Reply:
Ah nevermind, I was testing on 2.7, the error is different on 3.0+.

Ok please disregard that comment.

Reply:
Actually I just remembered there is a syntax to refuse keyword args:

```ruby
>> def foo(buf, **nil); end
=> :foo
>> foo(1, **{bar: 2})
(irb):16:in `<main>': no keywords accepted (ArgumentError)
>> foo(1, bar: 2)
(irb):17:in `<main>': no keywords accepted (ArgumentError)
```

I think I'd prefer to delegate that argument paring work to Ruby if possible. What do you think?

Reply:
That worked!